admin/users/views.py

```diff
@@ -263,6 +265,66 @@ def get_object(self, queryset=None):
         return serialize_user(User.load(self.kwargs.get('guid')))
 
 
+class UserWorkshopFormView(OSFAdmin, FormView):
+    form_class = WorkshopForm
+    object_type = 'user'
+    template_name = 'users/workshop.html'
+
+    def form_valid(self, form):
+        csv_file = form.cleaned_data['document']
+        final = self.parse(csv_file)
+        response = HttpResponse(content_type='text/csv')
+        response['Content-Disposition'] = 'attachment; filename="workshop.csv"'
+        writer = csv.writer(response)
+        for row in final:
+            writer.writerow(row)
+        return response
+
+    @staticmethod
+    def parse(csv_file):
+        final = []
+        for i, temp_line in enumerate(csv_file):
+            try:
+                line = temp_line.strip().split(',')[:-1]
+            except UnicodeDecodeError as e:
```

Comment:
Even parsing the csv file directly doesn't work. I think I will stick to catching errors and testing.